### PR TITLE
[MRG] Kit2fiff: reorder marker points

### PR DESCRIPTION
**Pull request by christianbrodbeck**
Minor new feature, a button to change the order of HPI marker points in case they were applied to the subject in the wrong order.

### CODE REVIEW

File: mne/gui/_marker_gui.py [hunk 1]
```diff
@@ -67,13 +67,36 @@
...
+    is_ok = Property(Bool, depends_on='order')
+
+    view = View(
+        Item('order', label='New order (five space delimited numbers)'),
+        buttons=[CancelButton, Action(name='OK', enabled_when='is_ok')])
+
+    def _get_index(self):
+        try:
+            return [int(i) for i in self.order.split()]
+        except:
```
Comment (larsoner) [inline mne/gui/_marker_gui.py]: always at least `except Exception` as `except:` will catch even `SystemExit`

**Comment by larsoner** [comment]
Newline contains whitespace

https://travis-ci.org/mne-tools/mne-python/jobs/283656260#L3084

**Comment by teonbrooks** [comment]
just tested it but I'm getting the following error:
```
2017-10-05 14:54:32.312 python[29188:5329462] modalSession has been exited prematurely - check for a reentrant call to endModalSession:
```

**Comment by christianbrodbeck** [comment]
I see that message too, but it does not stop it from working – is it associated with a disfunction for you?

**Comment by larsoner** [comment]
Have you tried using our shiny new experimental 3.6 environment? You could
look for my "env" PR (on mobile so linking is hard) and see if it has the
same problem

**Comment by teonbrooks** [comment]
> I see that message too, but it does not stop it from working – is it associated with a disfunction for you?

I couldn't render a view of the headshape. didn't know if this was the cause.

**Comment by christianbrodbeck** [comment]
@teonbrooks is this a general problem, or something that works on master but not on this PR?

**Comment by teonbrooks** [comment]
I explored it on the master branch and I discovered that in order to view the headshape, I had to press one of the view button. it works the same with this PR as well so that's fine.

one question, the switch left/right button, is this just a mirror of the coordinate positions?

LGTM. thanks @christianbrodbeck

**Comment by christianbrodbeck** [comment]
Yeah it’s been like this since the beginning, you have to do something to get it to draw the first time (dragging on the screen is enough), I put it down to a mayavi quirk, maybe there is a way to force a draw?

And no, mirroring coordinate positions would also chang ethe central points. Switch left-right switches the two lateral point pairs, i.e. if you accidentally acquire left before right. It was the first error I encountered and seemed to be the most likely to happen, but it is technically redundant with the other re-ordering, and if you think it is distracting we can remove it.

I don’t think you would ever want to mirror the coordinates, as the coordinates will always be in the proper coordinate system, just with different order (if the experimenter messes up the marker coils).

**Comment by teonbrooks** [comment]
yeah, I think that the reordering alone should be sufficient, but if you think the left/right switch happens often enough that the button would be useful, then I would leave it.

**Comment by larsoner** [comment]
> I put it down to a mayavi quirk, maybe there is a way to force a draw?

Have you tried just `mlab.draw(figure=f)`? It might be enough. If not, you can try what `PySurfer` [does](https://github.com/nipy/PySurfer/blob/master/surfer/viz.py#L174), though I doubt it will help here.

**Comment by christianbrodbeck** [comment]
@larsoner the issue is capturing the right events – the redraw ought to happen when you load data, which makes the relevant objects visible, but triggering a PySurfer-like redraw like `self.mrk1_obj.on_trait_change(self.on_fid_point_change_hduas, 'visible')` does not solve it. But that's probably not the whole issue either since even when I disable invisibility the points do not appear in the right locations when loading the object...

**Comment by larsoner** [comment]
@christianbrodbeck do you plan on making any changes after the discussion with @teonbrooks or is this still good to go?

**Comment by christianbrodbeck** [comment]
Since it's a button and not API I would leave it for now, can always be removed if we need more space...

**Comment by larsoner** [comment]
Thanks @christianbrodbeck

**Pull request merged**